src/lib.rs

```diff
@@ -465,6 +469,32 @@ mod tests {
         fn esc_dispatch(&mut self, _params: &[i64], _intermediates: &[u8], _ignore: bool, _byte: u8) {}
     }
 
+    #[derive(Default)]
+    struct DcsDispatcher {
+        dispatched_dcs: bool,
+        params: Vec<i64>,
+        c: Option<char>,
+        s: Vec<u8>,
+    }
+
+    impl Perform for DcsDispatcher {
+        fn print(&mut self, _: char) {}
+        fn execute(&mut self, _byte: u8) {}
+        fn hook(&mut self, params: &[i64], _intermediates: &[u8], _ignore: bool, c: char) {
+            self.c = Some(c);
+            self.params = params.to_vec();
+        }
+        fn put(&mut self, byte: u8) {
+            self.s.push(byte);
+        }
+        fn unhook(&mut self) {
+            self.dispatched_dcs = true;
+        }
+        fn osc_dispatch(&mut self, _params: &[&[u8]]) { }
+        fn csi_dispatch(&mut self, _params: &[i64], _intermediates: &[u8], _ignore: bool, _c: char) {}
+        fn esc_dispatch(&mut self, _params: &[i64], _intermediates: &[u8], _ignore: bool, _byte: u8) {}
+    }
+
```

Comment:
The only actual issue I have found with this PR is this double newline right here. Other than that, things are looking good to me.

Reply:
fixed!